nix/nix-build.sh

```diff
@@ -32,14 +32,30 @@ then
     nix-shell --pure $SHELL_NIX_FILE
 else
     ARGS=(--arg crates "${CRATES}" --arg singleStep "${SINGLE_STEP}" --arg rustflags "${RUSTFLAGS}")
-    nix-build --max-jobs auto $SHELL_NIX_FILE "${ARGS[@]}"
+
+    # first we download all dependencies
+    echo fetching depedencies...
+    CARGO_HOME="$(realpath ~/cargo-home)"
+    nix-shell --pure --run "CARGO_HOME=$CARGO_HOME cargo fetch --locked --offline || CARGO_HOME=$CARGO_HOME cargo fetch --locked"
```

Comment:
I think we might want to split this into two commands with more verbose output to a user, just in case one of them fail, so the user is not confused with an output

Reply:
Changed.